### PR TITLE
HTTP basic auth fix

### PR DESCRIPTION
**Pull request by cjsewell**
It appears with some apache/php configurations, the HTTP Basic auth header created by the .htaccess rule from [this fix](https://github.com/silverstripe/silverstripe-installer/pull/78) could exist in REDIRECT_HTTP_AUTHORIZATION or HTTP_AUTHORIZATION. 

Currently only HTTP_AUTHORIZATION is checked, this pull request fixes it to check both.
Also added a little documentation.

See [When setting environment variables in Apache RewriteRule directives, what causes the variable name to be prefixed with “REDIRECT_”?](http://stackoverflow.com/questions/3050444/when-setting-environment-variables-in-apache-rewriterule-directives-what-causes#answer-10128290) for more information about Server vars being prfixed with REDIRECT_

### CODE REVIEW

File: security/BasicAuth.php [hunk 1]
```diff
@@ -50,9 +50,19 @@ public static function requireLogin($realm, $permissionCode = null, $tryUsingSes
...
 
+                /*
+                 * Enable HTTP Basic authentication workaround for PHP running in CGI mode with Apache
+                 * Depending on server configuration the auth header may be in HTTP_AUTHORIZATION or
+                 * REDIRECT_HTTP_AUTHORIZATION
+                 * 
+                 * The follow rewrite rule must be in the sites .htaccess file to enable this workaround
+                 * RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
+                 */
+		$authHeader = isset($_SERVER['HTTP_AUTHORIZATION']) ? 'HTTP_AUTHORIZATION' :
```
Comment (tractorcow) [inline security/BasicAuth.php]: Instead of getting the header name to later retrieve from $_SERVER should we just get the value here?

**Comment by cjsewell** [comment]
Updated as @tractorcow suggested. . . . Far cleaner!

File: security/BasicAuth.php (hunk 1 above)
Comment (cjsewell) [inline security/BasicAuth.php]: Indeed! Don't ask why I implemented it the way I did?

File: security/BasicAuth.php [hunk 2]
```diff
@@ -50,9 +50,19 @@ public static function requireLogin($realm, $permissionCode = null, $tryUsingSes
...
+                /*
+                 * Enable HTTP Basic authentication workaround for PHP running in CGI mode with Apache
+                 * Depending on server configuration the auth header may be in HTTP_AUTHORIZATION or
+                 * REDIRECT_HTTP_AUTHORIZATION
+                 * 
+                 * The follow rewrite rule must be in the sites .htaccess file to enable this workaround
+                 * RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
+                 */
+		$authHeader = isset($_SERVER['HTTP_AUTHORIZATION']) ? $_SERVER['HTTP_AUTHORIZATION'] :
+			      isset($_SERVER['REDIRECT_HTTP_AUTHORIZATION']) ? $_SERVER['REDIRECT_HTTP_AUTHORIZATION'] : null;
```
Comment (tractorcow) [inline security/BasicAuth.php]: I'd probably bracket these just to be sure operator precedence doesn't screw the logic up.

http://stackoverflow.com/questions/6224330/understanding-nested-php-ternary-operator

Comment (cjsewell) [inline security/BasicAuth.php]: Makes sense. Updated!

Comment (cjsewell) [inline security/BasicAuth.php]: Bit odd, I have updated as sugged as you can see [here]https://github.com/guru-digital/silverstripe-framework/blob/REDIRECT_fix/security/BasicAuth.php#L61).

However the pull request diff hasnt updated....?

Comment (cjsewell) [inline security/BasicAuth.php]: Seems to have been updated now, so the diff now shows the brackets.

**Comment by tractorcow** [comment]
Mergled.

**Comment by cjsewell** [comment]
Excellent. Thanks. Any idea when the next release will be published?

**Comment by tractorcow** [comment]
Probably 3.1.12 sorry :)

**Pull request merged**